src/ui/buttons/buttonDesigns/resizeButtonShowLabelText.jsx

```diff
@@ -0,0 +1,196 @@
+    const { ANIMATION_CONTAINER, ANIMATION_LABEL_CONTAINER, PAYPAL_BUTTON_LABEL } = configuration.cssClasses;
+    const { min, smallMax, mediumMax } = configuration;
+    // get the animation main container to force specificity( in css ) and make sure we are running the right animation
+    const designContainer = (document && document.querySelector(`.${ ANIMATION_CONTAINER }`)) || null;
+    if (!designContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const designContainerWidth = designContainer.offsetWidth;
+    if (designContainerWidth < min || designContainerWidth > mediumMax) {
+        // remove label element from dom
+        designContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    const topPositionBlueLayer = designContainerWidth === mediumMax ? '26%' : '25%';
+
+    if (designContainerWidth >= smallMax) {
+        labelFontSize =  designContainerWidth === mediumMax ? 12 : 11;
+    }
+    
+    let buttonHeight = designContainer.offsetHeight;
+    if (designContainerWidth < smallMax) {
+        buttonHeight += 1;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalLabelContainer = designContainer.querySelector(`.${ PAYPAL_BUTTON_LABEL }`) || null;
+    const labelStylesObject = (paypalLabelContainer && (paypalLabelContainer.currentStyle || window.getComputedStyle(paypalLabelContainer))) || null;
+    const marginLabelContainer = (labelStylesObject && labelStylesObject.marginRight) || null;
+    return {
+        buttonHeight,
+        designContainer,
+        labelFontSize,
+        marginLabelContainer,
+        paypalLabelContainerElement: paypalLabelContainer,
+        topPositionBlueLayer
+    };
+};
+
+export const resizeButtonShowLabelTextAnimation = function (designProps : AnimationProps, configuration : Object) : void | null {
+    const { mediumMax, min, runOnce } = configuration;
+    const { ANIMATION_LABEL_CONTAINER, ANIMATION_CONTAINER, DOM_READY, PAYPAL_LOGO } = configuration.cssClasses;
+    const { buttonHeight, designContainer, paypalLabelContainerElement, labelFontSize, marginLabelContainer, topPositionBlueLayer } = designProps;
+    const timesToRunAnimation = runOnce ? '2' : 'infinite';
+    const blueLayerPosition = Math.round(parseFloat(marginLabelContainer)) + 1;
+    const animations = `
+        .${ DOM_READY } .${ ANIMATION_CONTAINER } img.${ PAYPAL_LOGO }-paypal{
+            animation: 4s move-logo-to-left-side 1s ${ timesToRunAnimation } alternate;
+            position:fixed;
+            transform:translateX(-50%);
+        }
+
+        .${ ANIMATION_CONTAINER } .blue-layer {
+            width: 1%;
+            height: ${ buttonHeight }px;
+            background-color: rgb(43,114,235);
+            position: fixed;
+            transform: translateY(-${ topPositionBlueLayer });
+            right: -${ blueLayerPosition }px;
```

Comment:
Is there a better way than calculating a pixel or percentage value every time? If not it's fine, but that would be cleaner

Reply:
we are not calculating this every time, this is just calculated one time ( when button has rendered ). and this is the initial position where we place the blue layer.